### ISSUE TITLE
Add searchbox/filter to save/load dialogue

### ISSUE THREAD

**Issue by ImperialAlex** [ImperialAlex/XLA_FixedArsenal#22]
Originally [requested](http://forums.bistudio.com/showthread.php?184838-quot-Fixed-quot-Arsenal-an-Arsenal-improving-workaround&p=2927279#post2927279) by [Tobinator](http://forums.bistudio.com/member.php?183956-Tobinator):
> Is it possible to add folders to the save function? Can you maybe try to add that function?

Since I don't feel comfortable tackling such a large UI modification on my own, I'd need a willing contributor with experience in UI who could work on this.

EDIT: Changed to search/filter box as suggested by @saintblood 
If somebody with the necessary skills wants to take a shot at folders they're still more than welcome to :)

**Comment by saintblood** [comment]
wouldnt be that beautiful but maybe even more of use... a searchbox could be added to the loading-dialog filtering the list below.

that way users could add certain patterns to their loadouts (ex. "US-OCP-Medic light") and filter the loadouts according to what they want to find.


should be easier to implement then making a folderbased system ++ more friendly to update if BI change something

**Comment by ImperialAlex** [comment]
Good point! I can probably manage a search box :)

**Comment by saintblood** [comment]
constructing the search-function flexible would be a good thing... it could maybe be implemented for item-lists or a global item-search (as a future plan.. if it can be implemented here it should be not much difficult to implement it there)

back to the topic... some thoughts:
* case sensitive - should not be necessary ([find](https://community.bistudio.com/wiki/find) however is case sensitive, resulting in the need of [toLower](https://community.bistudio.com/wiki/toLower)
* wildcards - would be nice to have (ex. ```us*medic*``` to get all kinds of us-medics)
* sorting - i believe the save/load dialog is not sorted in alphabetic order (or did BI change that?).. since human brains work better that way it would be nice to implement that as well
* multiuse of function - ability to search classnames and names of items

one more point against folderstructure.. as much as i like the idea.. it would be bad
XLA and BIS saved loadouts are compatible (or at least have been.. has it changed?), meaning each can load the other, since folders would touch the format they are saved it would break compatibility (or it would be a perverted construct in phonebookstyle which adds new vars to the userfile containing the info which loadout is saved in which folder - imo the userfile is cluttered up enough already)

**Comment by ImperialAlex** [comment]
Folders wouldn't necessarily be that much of a clusterf.... The "save inventory" function can append arbitrary custom data at the end. Earlier versions of the mod (not sure if those were ever released or got scrapped while still WIP) actually used that to implement a "favourites" tab in the save/load screen (a simply boolean flag was added to the custom field in the loadout data). Folders would still be tedious to implement, though ;)  Oh and yes, inventory is still compatible/the same between the arsenals. It's the only point of contact between XLA and BIS arsenal.

I've looked at wildcards already for the force-Replace feature (#16) but it looks like it'd require me to include a library/extension for RegExp's to pull it off. Since this is *just* a little arsenal mod I've decided against that for now.

Sorting: Oh, that's a good question. I guess that's an easy fix if it isn't already in lexicographic order.

I'm not quite sure what you mean by multiuse: If I were to implement a search/filter function for the save/load screen I would have probably have it filter base on the name of the saved loadout.
Adding a general search to the arsenal itself would be much much harder and probably not worth it (most people suffer from weapon-duplication because of dependencies. massi/CUP/HLC together can quickly end up with dozens of e.g. AK's

**Comment by saintblood** [comment]
i meant that the function should be of multiple use.. especially with wildcards in mind so you dont need a function to filter load, one for weapons, one for uniforms.. etc.. just one big fancy search function that could handle all that was just a thought about the future ;)

but since wildcards would not be possible, at least not with a huge workarround.. drop it, search for loadouts is fine enough :)


if i might ask.. why did you drop the favourites tab? would have loved that one.. much.. not going to 16 different versions of m4, m16 or ak would be soooooo nice

**Comment by ImperialAlex** [comment]
I dropped it because the 'feature creep' with that branch was getting to unmanageable levels. Vanilla kept getting updated and I was developing slower than BI was putting out changes. In the end I decided to scrap the entire branch and restart from vanilla which had been updated twice in the meantime. 
To keep things sane I decided to only implement the really necessary features... "favourites", along with mission-defined loadouts just weren't essential enough to make it in.
